cni/pkg/ambient/net.go

```diff
@@ -297,3 +302,39 @@ func (s *Server) DelPodFromMesh(pod *corev1.Pod, event controllers.Event) {
 func SetProc(path string, value string) error {
 	return os.WriteFile(path, []byte(value), 0o644)
 }
+
+func (s *Server) cleanStaleIPs(stales sets.Set[string]) {
+	log.Infof("Ambient stale Pod IPs to be cleaned: %s", stales)
+	switch s.redirectMode {
+	case IptablesMode:
+		for ip := range stales {
+			delIPsetAndRoute(ip)
+		}
+	case EbpfMode:
+		for ip := range stales {
+			if err := s.delPodEbpfOnNode(ip, false); err != nil {
+				log.Errorf("failed to cleanup POD(%s) ebpf: %v", ip, err)
+			}
+		}
+	}
+}
+
+func (s *Server) cleanupPodsEbpfOnNode() error {
+	if s.ebpfServer == nil {
+		return fmt.Errorf("uninitialized ebpf server")
+	}
+	for _, ns := range s.namespaces.List(
+		metav1.NamespaceAll, klabels.Set{pconstants.DataplaneMode: pconstants.DataplaneModeAmbient}.AsSelector()) {
+		namespace := ns.GetName()
+		for _, pod := range s.pods.List(namespace, klabels.Everything()) {
+			log.Infof("cleanup Pod %s in %s", pod.Name, namespace)
+			if err := s.delPodEbpfOnNode(pod.Status.PodIP, true); err != nil {
```

Comment:
This will be called on all pods -- but we may have some that explicitly opt out with annotation. So we should either filter those, or this needs to be sure to gracefully handle this.